src/expressions/util/Bucket.ts

```diff
@@ -78,3 +85,54 @@ export function intersectBuckets(bucket1: Bucket | null, bucket2: Bucket | null)
 	// Expression will never match any nodes
 	return 'empty';
 }
+
+/**
+ * Determine the union between the two passed buckets. The union is the common denominator between buckets
+ *
+ * Example: `null` ∪ `name-div` = `null`
+ * Example: `name-p` ∪ `name-div` = `type-1-or-type-2`
+ * Example: `type-1` ∪ `name-p` = `type-1`
+ * Example: `type-1` ∪ `empty` = `type-1`
+ *
+ * @param  bucket1 - The first bucket to check
+ * @param  bucket2 - The second bucket to check
+ *
+ * @returns The union of the two buckets.
+ */
+export function unionBucket(bucket1: Bucket | null, bucket2: Bucket | null): Bucket | null {
```

Comment:
Nice, I think we can also use this in OrOperator (which currently has a simpler implementation).

Reply:
Done!